File: Python/sin.py
def factorial(x):
    f = 1
    for i in range(2, x+1):
        f *= i
    return f
    
def sin(angle, precision):
    result = angle
    i = 3
    for j in range(0, precision):
        if j % 2 != 0:
            result += angle**i / factorial(i)
        else:
            result -= angle**i / factorial(i)
        i += 2
    return result


PI = 3.14159265359   
print("%.3f" % sin(PI/2, 10))

File: C/sin.c
#include <stdio.h>
#include <math.h>

#define PI 3.14159265359

typedef long long unsigned int ll_uint;

ll_uint factorial(ll_uint x)
{
    ll_uint f = 1;
    for (ll_uint i = x; i > 1; i--)
        f *= i;
    return(f);
}

long double Sin(const long double angle, ll_uint precision)
{
    long double result = angle;
    for (ll_uint i = 3, j = 0; j < precision; i += 2, j++)
    {
        if (j % 2 != 0)
        {
            result += pow(angle, i) / factorial(i);
        }
        else
        {
            result -= pow(angle, i) / factorial(i);
        }
    }
    return(result);
}

int main()
{
    printf("%.3Lf\n", Sin(PI/2, 10));
    return 0;
}
